src/Servers/Kestrel/Core/src/Internal/Http2/Http2MessageBody.cs

```diff
@@ -51,6 +52,7 @@ public override void Reset()
     {
         base.Reset();
         _readResult = default;
+        ExtendedConnect = _context.IsExtendedConnectRequest;
```

Comment:
Is this correct?

Reply:
Talked offline: yes, basically the context gets reset separately and will have the correct value when the Http2MessageBody is reset

Reply:
Isn't it always false here though? Let's not make it any more convoluted than necessary.

Reply:
Could make `ExtendedConnect` virtual, and override in `Http2MessageBody` to wrap `Context.IsExtendedConnectRequest`. Simpler, safer, but very small perf hit of virtual property call.

Reply:
IsExtendedConnectRequest is initialized before Http2MessageBody is reset.